prompt-service/src/unstract/prompt_service/variable_extractor/prompt_variable_service.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import re
+from typing import Any
+
+from utils.request import HTTPMethod, make_http_request
+
+from .constants import VariableConstants, VariableType
+
+logger = logging.getLogger(__name__)
+
+
+class VariableService:
+
+    @staticmethod
+    def replace_static_variable(
+        prompt: str, structured_output: dict[str, Any], variable: str
+    ) -> str:
+        output_value = VariableService.check_static_variable_run_status(
+            structure_output=structured_output, variable=variable
+        )
+        static_variable_marker_string = "".join(["{{", variable, "}}"])
+        if not output_value:
+            return prompt
+        replaced_prompt: str = VariableService.replace_generic_string_value(
+            prompt=prompt, variable=static_variable_marker_string, value=output_value
+        )
+
+        return replaced_prompt
+
+    @staticmethod
+    def check_static_variable_run_status(
+        structure_output: dict[str, Any], variable: str
+    ) -> Any:
+        output = None
+        try:
+            output = structure_output[variable]
+            return output
+        except KeyError:
+            logger.warn(
+                f"Prompt with {variable} is not executed yet."
+                " Unable to replace the variable"
+            )
+            return output
+
+    @staticmethod
+    def replace_generic_string_value(prompt: str, variable: str, value: str) -> str:
+        replaced_prompt = re.sub(variable, value, prompt)
+        return replaced_prompt
+
+    @staticmethod
+    def identify_variable_type(variable: str) -> VariableType:
+        variable_type: VariableType
+        pattern = re.compile(VariableConstants.DYNAMIC_VARIABLE_URL_REGEX)
+        if re.findall(pattern, variable):
+            variable_type = VariableType.DYNAMIC
+        else:
+            variable_type = VariableType.STATIC
+        return variable_type
+
+    @staticmethod
+    def replace_dynamic_variable(prompt: str, variable: str) -> str:
+        url = re.search(VariableConstants.DYNAMIC_VARIABLE_URL_REGEX, variable).group(0)
+        data = re.findall(VariableConstants.DYNAMIC_VARIABLE_DATA_REGEX, variable)[0]
```

Comment:
This data could be another static variable right? @harini-venkataraman

Reply:
In that case, we may have to replace it similar to what we have done for replace_static_variable

Reply:
@gaya3-zipstack `data` is the static variable, which will be sent as an attribute to POST API

Reply:
The entire variable will be replaced here  with the response @gaya3-zipstack

Reply:
Ok as long as something like {{[Prev_prompt_key]https://x.y.com}} is taking care pf replacing Prev_prompt_key with the response of the prev prompt and then sending it to the URL for a POST, I am ok.